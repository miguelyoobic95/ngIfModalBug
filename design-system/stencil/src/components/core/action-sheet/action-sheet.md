---
name: Action Sheet
category: Core Components
---
An Action Sheet is a dialog that lets the user choose from a set of options. It appears on top of the app's content, and must be manually dismissed by the user before they can resume interaction with the app. 

See the Modal controller for details on how to implement

The following interface is used:

```tsx 
export interface IActionSheetButton  {
    text?: string;
    icon?: string;
    disabled?: boolean;
    cssClass?: string;
}

export interface IActionSheet  {
    heading?: string;
    buttons: IActionSheetButton[];
}
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`heading`|string|   |the title of the action sheet|
|`buttons[]`|IActionSheetButton[]|  |entry to create the action sheet heading and buttons|

```yoo-action-sheet.html
<yoo-modal-controller></yoo-modal-controller>
<div style="padding:10px;">
    <yoo-button text="Trigger Action Sheet" class="accent squared" onclick="generate()"></yoo-button>
</div>
```

```yoo-action-sheet.js
    var modalCtrl = document.querySelector('yoo-modal-controller');
    var htmlElement = document.createElement('div');
    htmlElement.innerHTML = `<div class="test">This is the test text</div>`;
    var actionSheetProps = { heading: 'Action sheet', buttons: [{text: 'Action 1', cssClass: "accent"}, {text: 'Action 2', icon: 'yo-fire', cssClass: "success"}, {text: 'Action 3', icon: 'yo-fire', cssClass: "info"}, {text: 'Disabled Action 4', icon: 'yo-fire', cssClass: "accent", disabled: "true"}] };

    function generate() {
    modalCtrl.generateActionSheet(actionSheetProps);
    modalCtrl.showActionSheet();
  }
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-action-sheet accent|
|Dark|`dark`|yoo-action-sheet dark|
|Success|`success`|yoo-action-sheet success|
|Danger|`danger`|yoo-action-sheet danger|
|Info|`info`|yoo-action-sheet info|
|Warning|`warning`|yoo-action-sheet warning|
|Gradient Accent|`gradient-accent`|yoo-action-sheet gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-action-sheet gradient-grey|
|Gradient Success|`gradient-success`|yoo-action-sheet gradient-success|
|Gradient Danger|`gradient-danger`|yoo-action-sheet gradient-danger|
|Gradient Info|`gradient-info`|yoo-action-sheet gradient-info|
|Gradient Warning|`gradient-warning`|yoo-action-sheet gradient-warning|

## Events

|Attr|Description|
|---|---|---|
|`actionSelected`|event triggered when a button supplied to the action sheet has been clicked. Returns a string value equal to the title of the button, will also dismiss the action sheet|
|`actionSheetClosed`|event triggered when the menu is closed|