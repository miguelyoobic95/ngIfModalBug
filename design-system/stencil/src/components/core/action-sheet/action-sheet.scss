
@mixin position-size() {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 3.125rem;
    padding: 0 1rem;
}

@mixin color($color) {
    :host(.#{$color}) {
        .outer-container {
            .top-container {
                .heading-container {
                    color: cssvar($color);
                }
            }
        }
    }
}

:host() {
    width: 100%;
    .outer-container {
        font-size: 0.85rem;
        width: 100%;
        padding: 0.75rem;
        .top-container {
            justify-content: center;
            background-color: cssvar(light);
            border-radius: 0.8125rem;      
            width: 100%;
            .heading-container {
                color: cssvar(stable);
                @include position-size();
                height: 2.875rem;
                font-size: 0.9rem;
                font-weight: 400;                
            }
            .inner-container {
                height: 3.625rem;
                font-size: 1.25rem;
                border-top: 1px solid cssvar(stable-10);
                color: cssvar(dark); 
                @include position-size();
                font-weight: 400;
            }
        }
        .bottom-container {
            background-color: transparent;        
            padding-top: 0.5rem;
            width: 100%;
            .cancel-container {
                font-size: 1.25rem;                
                background-color: cssvar(light);
                height: 3.5625rem;        
                border-radius: 0.8125rem;
                font-weight: 400;                
                color: cssvar(danger);
                @include position-size();            
            }
        }
    }
}

@include color(accent);
@include color(success);
@include color(danger);
@include color(info);
@include color(warning);
@include color(dark);
@include color(stable);