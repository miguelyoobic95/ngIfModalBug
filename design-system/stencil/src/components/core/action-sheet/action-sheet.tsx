import { Component, Prop, Event, EventEmitter, Element } from '@stencil/core';
import { IActionSheetButton } from '@shared/interfaces';
import { setAnimation } from '../../../utils/anim';

@Component({
    tag: 'yoo-action-sheet',
    styleUrl: 'action-sheet.scss',
    scoped: true
})
export class YooActionSheetComponent {

    @Prop() heading: string;
    @Prop() buttons: IActionSheetButton[] = [];

    @Event() actionSelected: EventEmitter<string>;
    @Event() actionSheetClosed: EventEmitter<boolean>; // catch and re send close modal event

     @Element() host: HTMLStencilElement;

    closeActionSheet(): void {
        setAnimation('slide_down', this.host.querySelector('div'), {open: false, up: false});
        this.actionSheetClosed.emit(true);
    }

    onButtonClick(heading: string, disabled: boolean): void {
        if (!disabled) {
            this.actionSelected.emit(heading);
            this.closeActionSheet();
        }
    }

    componentDidLoad() {
        setAnimation('slide_down', this.host.querySelector('div'), {open: true});
    }

    render(): JSX.Element {
        return (
            <yoo-modal has-header="false" class="action-sheet" onClosed={() => this.closeActionSheet()}>
                <div  attr-layout="column">
                <div class="outer-container">
                    <div class="top-container" attr-layout="column">
                        <div class="heading-container">
                            {this.heading}
                        </div>
                        {this.buttons.map((button) =>
                        <div class="inner-container" onClick={() => this.onButtonClick(button.text, button.disabled)}>
                                {button.text}
                                {button.icon ? (<i class={button.icon}></i>) : null}
                            </div>
                        )}
                    </div>
                    <div class="bottom-container">
                        <div class="cancel-container" onClick={() => this.closeActionSheet()}>
                            Cancel
                        </div>
                    </div>
                </div>
                </div>
            </yoo-modal>
        );
    }
}