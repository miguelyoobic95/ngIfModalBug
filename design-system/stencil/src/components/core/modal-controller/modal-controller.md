---
name: Modal Controller
category: Core Components
---

The Modal Controller acts as a placeholder for a modal to be inserted. Different modals may be inserted into the same controller. The controller provides the backdrop for that modal so that the content behind it is greyed out. It has no attributes.

The modal is constructed by calling the `generateModal(props: IModalEntry)` method which receives a Modal Entry as an argument.

### Displaying Modals with the ModalController

```yoo-modal-controller.html
    <p>This is a random text in a random page</p>
    <p> Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star systems is in dispute. Hoping to resolve the matter with a blockade of deadly battleships, the greedy Trade Federation has stopped all shipping to the small planet of Naboo. While the Congress of the Republic endlessly debates this alarming chain of events, the Supreme Chancellor has secretly dispatched two Jedi Knights, the guardians of peace and justice in the galaxy, to settle the conflict....</p>

    <yoo-modal-controller></yoo-modal-controller>
    <yoo-button text="Trigger Modal" class="accent squared" onclick="generate(modalProps1)"></yoo-button>
    <yoo-button text="Trigger Fullscreen Modal" class="accent squared" onclick="generate(modalProps2)"></yoo-button>
    <yoo-button text="Trigger Drawer Modal" class="accent squared" onclick="generate(modalProps3)"></yoo-button>
    <yoo-button text="Trigger Action Sheet" class="accent squared" onclick="generateActionSheet()"></yoo-button>
    <br/>
    <yoo-button text="Confirm Method" onclick="confirm()"></yoo-button>
    <yoo-button text="Populate Method" onclick="populateShow()"></yoo-button>
```

```yoo-modal-controller.js

  var modalCtrl = document.querySelector('yoo-modal-controller');

  var htmlElement = document.createElement('div');

  var modalProps1 = {heading: "Aligato ?", hasFooter : true, content: htmlElement,primaryButtonText: "Confirm", secondaryButtonText: "Cancel", animationName: "rotate_bottom_3d", withYooCtrl: "true"};
  var modalProps2 = {heading: "Aligato ?", hasFooter : true, content: htmlElement,primaryButtonText: "Confirm", secondaryButtonText: "Cancel", cssClass: "fullscreen", animationName: "fade_in_scale", withYooCtrl: "true"};
  var modalProps3 = {heading: "Aligato ?", hasFooter : true, content: htmlElement,primaryButtonText: "Confirm", secondaryButtonText: "Cancel", cssClass: "drawer", animationName: "slide_right", withYooCtrl: "true"};
  var actionSheetProps = { heading: 'Action sheet', buttons: [{text: 'Action 1', cssClass: "accent"}, {text: 'Action 2', icon: 'yo-fire', cssClass: "success"}, {text: 'Action 3', icon: 'yo-fire', cssClass: "info"}, {text: 'Disabled Action 4', icon: 'yo-fire', cssClass: "accent", disabled: "true"}] };

  function generate(modalProps) {
    modalCtrl.generateModal(modalProps);
    modalCtrl.show();
  }

  function confirm() {
    modalCtrl.confirm(true);
    modalCtrl.show();
  }

  function generateActionSheet() {
    modalCtrl.generateActionSheet(actionSheetProps);
    modalCtrl.showActionSheet();
  }

  function populateShow() {
    var modal = document.createElement('yoo-modal');

    htmlElement.innerHTML = `<div class="test">This text is added from populate Method</div>`;

    modal.heading = "Gutten Morgen";
    modal.hasFooter = true;
    modal.content = htmlElement;
    modal.classList.add("info");
    modal.withYooCtrl = true;

    modalCtrl.populateModal(modal);

    modalCtrl.show();
  }

```

### Displaying Alerts with the ModalController

```yoo-modal-controller-alert.html
    <p>This is a random text in a random page</p>
    <p> Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star systems is in dispute. Hoping to resolve the matter with a blockade of deadly battleships, the greedy Trade Federation has stopped all shipping to the small planet of Naboo. While the Congress of the Republic endlessly debates this alarming chain of events, the Supreme Chancellor has secretly dispatched two Jedi Knights, the guardians of peace and justice in the galaxy, to settle the conflict....</p>

    <yoo-modal-controller></yoo-modal-controller>
    <yoo-button text="Trigger Toast Alert" class="accent squared" onclick="generate(alertProps1)"></yoo-button>
    <yoo-button text="Trigger Toast Alert" class="success squared" onclick="generate(alertProps2)"></yoo-button>
    <yoo-button text="Trigger Toast Alert" class="gradient-warning squared" onclick="generate(alertProps3)"></yoo-button>
```

```yoo-modal-controller-alert.js

  var modalCtrl = document.querySelector('yoo-modal-controller');
  var alertProps1 = {heading: "Toast Alert", text: "An alert has been launched !", closeable: true, cssClass: "toast accent", animationName: "slide_right", withYooCtrl: "true"};
  var alertProps2 = {heading: "Toast Alert", text: "An alert has been launched !", closeable: true, cssClass: "toast success", animationName: "slide_down", withYooCtrl: "true"};
  var alertProps3 = {heading: "Toast Alert", text: "An alert has been launched !", closeable: true, cssClass: "toast warning", animationName: "shake", withYooCtrl: "true"};

  function generate(alertProps) {
    modalCtrl.generateAlert(alertProps);
    modalCtrl.showAlert();
  }

```

## Methods

|Name|Parameters|Description|
|---|---|---|
|`populateModal`|`modal: HTMLElement`| Set the content to be displayed|
|`generateModal`|`props: IModalEntry`| Set the content to be displayed to a Modal created using the ModalEntry parameter|
|`generateActionSheet`|`props: ActionSheet`| Set the content to be displayed to a ActionSheet created using the ActionSheet parameter|
|`generateAlert`|`props: IAlertEntry`| Set the content to be displayed to an Alert created using the AlertEntry parameter|
|`confirm`|none| Set a confirm content to be displayed to a Modal created|
|`show`|none| Interrupt the current page and display content above|
|`showAlert`|none| Interrupt the current page and display content above|
|`closeModal`|none| Close the modal and placeholder|
|`closeAlert`|none| Close the alert|
|`closeActionSheet`|none| Close the action sheet|

### Vue JS Example, creating a modal with a vue component in it

```yoo-modal-controller-vue.html
    <p>This is a random text in a random page</p>
    <p> Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star systems is in dispute. Hoping to resolve the matter with a blockade of deadly battleships, the greedy Trade Federation has stopped all shipping to the small planet of Naboo. While the Congress of the Republic endlessly debates this alarming chain of events, the Supreme Chancellor has secretly dispatched two Jedi Knights, the guardians of peace and justice in the galaxy, to settle the conflict....</p>

    <yoo-modal-controller></yoo-modal-controller>
    <yoo-button text="Modal With Vue Component" class="accent squared" onclick="generate1()"></yoo-button>
      <div id="hello-vue">
        <h1>{{ msg }}</h1>
      </div>
```

```yoo-modal-controller-vue.js

  //this is a vue instance
  let vueElem = new Vue({
      //this targets the div id app
      el: '#hello-vue',
      data: {
          msg: 'This is a view component !' //this stores data values for 'name'
      }
  })

  var modalCtrl = document.querySelector('yoo-modal-controller');
  let htmlElement = vueElem.$el;

  var modalProps1 = {heading: "Aligato ?", hasFooter : true, content: htmlElement,primaryButtonText: "Confirm", secondaryButtonText: "Cancel"};

  function generate1() {
    modalCtrl.generateModal(modalProps1);
    modalCtrl.show();
  }
```