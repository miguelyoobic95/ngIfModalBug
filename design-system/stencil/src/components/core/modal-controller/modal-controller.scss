

:host {
    position: absolute;
    z-index: -1;
    display: flex;
    height: 100%;
    width: 100%;
    top: 0%;
    align-items: center;
    justify-content: center;
    .placeholder {
        z-index: -1;
        position: absolute;
        top: 0px;
        left: 0px;
        width: 100%;
        height: 100%;
        background-color: rgba(58, 67, 97, 0.5);
    }
    .transparent {
        height: 100%;
        width: 100%;
    }

    .alert-container {
        position: fixed;
        bottom: 20px;
        right: 20px;
    }
}
