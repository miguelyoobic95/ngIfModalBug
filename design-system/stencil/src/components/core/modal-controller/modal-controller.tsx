import { Component, Element, State, Method, Listen, Event, EventEmitter, Prop } from '@stencil/core';
import { IModalEntry, IAlertEntry, IActionSheet, ILanguageSelector } from '@shared/interfaces';
import { setAnimation } from '../../../utils/anim';

@Component({
    tag: 'yoo-modal-controller',
    styleUrl: 'modal-controller.scss',
    scoped: true
})
export class YooModalControllerComponent {

    @Prop() greyContent: boolean = true;

    @State() element: HTMLElement = document.createElement('yoo-modal');
    @State() isGreyedOut: boolean = false;
    @State() displayedAlert: number = 0;

    @Event() modalCtrlPrimaryButtonClicked: EventEmitter<boolean>;

     @Element() host: HTMLStencilElement;

    @Listen('closed')
    childClosed() {
        this.host.classList.add('out');
        // setTimeout(() => {this.isGreyedOut = false; }, 300);
        this.closeModal(true);
    }

    @Listen('actionSheetClosed')
    childActionClosed() {
        this.host.classList.add('out');
        // setTimeout(() => {this.isGreyedOut = false; }, 300);
        this.closeModal(true);
    }

    // Event transmission
    @Listen('modalPrimaryButtonClicked')
    primaryClick() {
        this.host.classList.add('out');
        this.modalCtrlPrimaryButtonClicked.emit(true); //event transmission
        //console.log('Primary Button Click on modal transmitted by modal Controller');
        this.closeModal(true);
    }

    @Listen('alertClosed')
    onAlertClosed(event: CustomEvent) {
        this.closeAlert(event.srcElement);
    }

    @Method()
    show() {
        this.isGreyedOut = true;
        this.host.setAttribute('style', 'z-index: 2000 !important;');
        const backDrop = this.host.querySelector('div');
        setAnimation('background_fade', backDrop, {open: true});
        this.host.querySelector('div').insertAdjacentElement('afterend', this.element);
    }

    @Method()
    showActionSheet() {
        this.isGreyedOut = true;
        this.host.setAttribute('style', 'z-index: 2000 !important; align-items: flex-end;');
        const backDrop = this.host.querySelector('div');
        setAnimation('background_fade', backDrop, {open: true});
        this.host.querySelector('div').insertAdjacentElement('afterend', this.element);
    }

    @Method()
    showAlert() {
        //this.show();
        this.host.querySelector('div.alert-container').appendChild(this.element);
        //this.isGreyedOut = false;
        this.displayedAlert = Math.max(0, this.displayedAlert - 1);
    }

    @Method()
    closeModal(sentFromModal: boolean) {
        const modal = this.host.querySelector('yoo-modal');
        const actionSheet = this.host.querySelector('yoo-action-sheet');
        const languageSelector = this.host.querySelector('yoo-language-selector');
        const backDrop = this.host.querySelector('div');
        setAnimation('background_fade', backDrop, {open: false});
        if (sentFromModal === false) {
            if (modal.animationName === 'sticky_up') {
                const padding = 16;
                const modalHeight = modal.clientHeight;
                setAnimation('sticky_up', modal, {open: false, distance: (((window.innerHeight / 2) + padding) - (modalHeight / 2)), modalHeight: modalHeight});
            } else {
                setAnimation(actionSheet ? 'slide_down' : (languageSelector ? 'fade' : modal.animationName), actionSheet ? actionSheet : (languageSelector ? languageSelector : modal), {open: false, up: false});
            }
        }
        setTimeout(() => {
            this.host.setAttribute('style', 'z-index: -1 !important;');
            modal.remove();
            if (actionSheet) { actionSheet.remove(); }
            if (languageSelector) { languageSelector.remove(); }
            this.isGreyedOut = false;
        }, 200);
    }

    @Method()
    closeAlert(alert) {
        if (alert) {
            setAnimation(alert.animationName, alert, {open: false});
            setTimeout(() => {
                alert.remove();
            }, 200);
        }
    }

    @Method()
    closeActionSheet() {
        let actionSheet = this.host.querySelector('yoo-action-sheet');
        if (actionSheet) {
            actionSheet.remove();
        }
    }

    @Method()
    populateModal(modal: HTMLElement) {
        this.element = modal;
    }

    @Method()
    generateModal(props: IModalEntry) {
        let m = document.createElement('yoo-modal');
        // m = setModalProps(m, props);
        m = Object.assign(m, props);
        m.className += props.cssClass;
        this.element = m;
    }

    @Method()
    generateAlert(props: IAlertEntry) {
        let newAlert = document.createElement('yoo-alert');
        // newAlert = setAlertProps(newAlert, props);
        newAlert = Object.assign(newAlert, props);
        newAlert.className += props.cssClass;
        this.element = newAlert;
    }

    @Method()
    generateActionSheet(props: IActionSheet) {
        let newActionSheet = document.createElement('yoo-action-sheet');
        newActionSheet = Object.assign(newActionSheet, props);
        newActionSheet.className += props.cssClass;
        this.element = newActionSheet;
    }

    @Method()
    generateLanguageSelector(props: ILanguageSelector) {
        let newLanguageSelector = document.createElement('yoo-language-selector');
        newLanguageSelector = Object.assign(newLanguageSelector, props);
        this.element = newLanguageSelector;
    }

    @Method()
    confirm(customController: boolean = false, cssClass = 'accent') {
        const content: HTMLElement = document.createElement('div');
        content.innerHTML = 'Please confirm the previous action';
        const confirmModalProps: IModalEntry = {
            heading: 'Action Required',
            headingIcon: null,
            hasHeader: true,
            hasFooter: true,
            footerText: null,
            content: content,
            primaryButtonText: 'Confirm',
            secondaryButtonText: 'Cancel',
            cssClass: cssClass,
            animationName: 'fade_in_scale',
            withYooCtrl: customController
        };
        this.generateModal(confirmModalProps);
    }

    render(): JSX.Element {
        return (
            [<div onClick={() => this.closeModal(false)} class={this.greyContent ? (this.isGreyedOut ? 'placeholder' : '') : 'transparent'}>
            </div>,
            <div class="alert-container">
            </div>]
        );

    }
}
