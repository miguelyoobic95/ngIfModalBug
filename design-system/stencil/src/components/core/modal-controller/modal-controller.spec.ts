import { TestWindow } from '@stencil/core/testing';
jest.mock('../../../utils/anim');
import { YooModalControllerComponent } from './modal-controller';
import { YooActionSheetComponent } from '../action-sheet/action-sheet';
import { YooAlertComponent } from '../alert/alert';
import { YooModalComponent } from '../modal/modal';

describe('YooModalControllerComponent', () => {
    let window;
    beforeEach(() => {
      window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooModalControllerComponent()).toBeTruthy();
    });

    describe('Generate Component to display', () => {
        it ('should create a single modal with header and footer', async () => {
            let element = new YooModalControllerComponent();
            element.generateModal({
                heading: 'Gutten Morgen',
                hasFooter: true
            });
            expect(element.element).toBeTruthy();

            let modal = await window.load({
                components: [YooModalComponent],
                html: element.element.outerHTML
            });
            expect(modal).toMatchSnapshot();
        });

        it ('should populate a single modal with header', async () => {
            let modalController = new YooModalControllerComponent();

            let element = document.createElement('yoo-modal');
            let htmlElement = document.createElement('div');
            htmlElement.innerHTML = `<div class="test">This is the test text</div>`;
            element.heading = 'heading';
            element.hasHeader = true;
            element.content = htmlElement;

            let modal = await window.load({
                components: [YooModalComponent],
                html: element.outerHTML
            });

            modalController.populateModal(modal);
            expect(modal).toMatchSnapshot();
        });

        it ('should generate a confirm modal', async () => {
            let modalController = new YooModalControllerComponent();

            modalController.confirm();


            let modal = await window.load({
                components: [YooModalComponent],
                html: modalController.element.outerHTML
            });

            expect(modal.className).toContain('accent');
            expect(modal).toMatchSnapshot();
        });

        it ('should create a single toast alert', async () => {
            let element = new YooModalControllerComponent();
            element.generateAlert({heading: 'Toast Alert',
                text: 'An alert has been launched !',
                closeable: true,
                cssClass: 'toast modal accent'});
            expect(element.element).toBeTruthy();

            let alert = await window.load({
                components: [YooAlertComponent],
                html: element.element.outerHTML
            });
            expect(alert).toMatchSnapshot();
        });

        it ('should create an action sheet', async () => {
            let element = new YooModalControllerComponent();
            element.generateActionSheet({
                heading: 'Action sheet',
                buttons: [{text: 'Action 1', cssClass: 'accent'}, {text: 'Action 2', icon: 'yo-fire', cssClass: 'success'}]
            });
            expect(element.element).toBeTruthy();

            let actionSheet = await window.load({
                components: [YooActionSheetComponent],
                html: element.element.outerHTML
            });
            expect(actionSheet).toMatchSnapshot();
        });
    });
    //TODO: Event test
  // describe('events', () => {
  //   let element, res, emitEvent;
  //   beforeEach( async () => {
  //     element = new YooModalControllerComponent();
  //     emitEvent = jest.fn(ev => { res = ev.detail; });
  //   });
  //   it ('should emit modalCtrlPrimaryButtonClicked', async () => {
  //     element.modalCtrlPrimaryButtonClicked = {
  //       emit: () => {}
  //     };
  //     const changedSpy = jest.spyOn(element.modalCtrlPrimaryButtonClicked, 'emit');
  //     element.primaryClick();
  //     expect(changedSpy).toHaveBeenCalledWith(24);
  //   });
  // });
});
