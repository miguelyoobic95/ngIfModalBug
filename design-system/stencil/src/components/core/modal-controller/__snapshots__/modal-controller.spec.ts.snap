// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooModalControllerComponent Generate Component to display should create a single modal with header and footer 1`] = `
<yoo-modal
  class="undefined hydrated"
  data-yoo-modal-host=""
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalControllerComponent Generate Component to display should create a single toast alert 1`] = `
<yoo-alert
  class="toast modal accent hydrated"
  data-yoo-alert-host=""
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        
        <span
          class="value"
          data-yoo-alert=""
        >
          
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;

exports[`YooModalControllerComponent Generate Component to display should create an action sheet 1`] = `
<yoo-action-sheet
  class="undefined hydrated"
  data-yoo-action-sheet-host=""
>
  
  <yoo-modal
    class="action-sheet"
    data-yoo-action-sheet=""
    has-header="false"
  >
    <div
      attr-layout="column"
      data-yoo-action-sheet=""
    >
      <div
        class="outer-container"
        data-yoo-action-sheet=""
      >
        <div
          attr-layout="column"
          class="top-container"
          data-yoo-action-sheet=""
        >
          <div
            class="heading-container"
            data-yoo-action-sheet=""
          >
            
          </div>
        </div>
        <div
          class="bottom-container"
          data-yoo-action-sheet=""
        >
          <div
            class="cancel-container"
            data-yoo-action-sheet=""
          >
            Cancel
          </div>
        </div>
      </div>
    </div>
  </yoo-modal>
</yoo-action-sheet>
`;

exports[`YooModalControllerComponent Generate Component to display should generate a confirm modal 1`] = `
<yoo-modal
  class="accent hydrated"
  data-yoo-modal-host=""
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalControllerComponent Generate Component to display should populate a single modal with header 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;
